# Generated by Django 5.0.14 on 2025-07-22 09:35

from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('agenda', '0009_alter_agenda_obra_alter_agendamento_data_criacao'),
        ('obra', '0009_alter_obra_status'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.RemoveField(
            model_name='agenda',
            name='entregador',
        ),
        migrations.RemoveField(
            model_name='agenda',
            name='obra',
        ),
        migrations.RemoveConstraint(
            model_name='agendamento',
            name='unique_montador_agendamento',
        ),
        migrations.AddField(
            model_name='agendamento',
            name='cancelado',
            field=models.BooleanField(default=False, verbose_name='Agendamento Cancelado?'),
        ),
        migrations.AddField(
            model_name='agendamento',
            name='status',
            field=models.CharField(choices=[('pendente', 'Pendente'), ('em_andamento', 'Em Andamento'), ('realizado', 'Realizado'), ('cancelado', 'Cancelado')], default='pendente', max_length=20, verbose_name='Status do Agendamento'),
        ),
        migrations.AlterField(
            model_name='agendamento',
            name='data_agendamento',
            field=models.DateTimeField(blank=True, null=True, verbose_name='Data de Agendamento'),
        ),
        migrations.AddConstraint(
            model_name='agendamento',
            constraint=models.UniqueConstraint(condition=models.Q(('data_agendamento__isnull', False)), fields=('montador', 'data_agendamento'), name='unique_montador_agendamento', violation_error_message='Esse montador já tem um agendamento para esse horário.'),
        ),
        migrations.DeleteModel(
            name='agenda',
        ),
    ]
